go/client/logclient.go

```diff
@@ -125,20 +125,21 @@ func (c *LogClient) fetchAndParse(uri string, res interface{}) error {
 		return err
 	}
 	resp, err := c.httpClient.Do(req)
-	var body []byte
-	if resp != nil {
-		body, err = ioutil.ReadAll(resp.Body)
-		resp.Body.Close()
-		if err != nil {
-			return err
-		}
-	}
 	if err != nil {
 		return err
 	}
-	if err = json.Unmarshal(body, &res); err != nil {
+	defer resp.Body.Close()
+	// Make sure everything is read, so http.Client can reuse the connection.
+	defer ioutil.ReadAll(resp.Body)
```

Comment:
is the point here that if json.NewDecoder...Decode doesn't run (or terminates prematurely) that the deferred 'ReadAll' gets to the end of the input regardless?

is that a noop in the case of JSON parsing out successfully?